### ISSUE TITLE
Add Astroplan dependencies for tests / docs

### ISSUE THREAD

**Issue by cdeil** [astroplanners/astroplan#6]
@bmorris3 - could you please update `setup.py`, `.travis.yml` `docs/rtd-pip-requirements` and maybe add a section to the docs about required and optional dependencies for Astroplan.

I think required should be:
* Numpy >=1.6
* Astropy >= 1.0
* https://pypi.python.org/pypi/pytz

Optional:
* matplotlib
* maybe for now pyephem or skyfield if you plan to use them to implement some functionality?

There's usually some issues getting the builds to run on travis-ci and readthedocs when changing the config, so I think it's worth splitting out into a separate PR instead of mixing it with code implementation PRs.

**Comment by bmorris3** [comment]
Do you have any suggestions for resources on how to update `.travis.yml`? Should we use the [recipe for installing missing dependencies with apt-get](http://docs.travis-ci.com/user/build-configuration/#Installing-Packages-Using-apt)?

**Comment by cdeil** [comment]
@bmorris3 – I think the existing `.travis.yml` file in this repo or in the Astropy or other affiiliated package repos are the best resources. Add to the existing `pip install` or `conda install` lines or put new lines using `conda install` or `pip install`.

**Comment by bmorris3** [comment]
I have a lame git question. (With lack of foresight) I made the commits with API updates on my master branch. To make a new branch with just the doc updates from this issue, I created a new branch, `git reset` back to last commit before I added the API, and added the new docs updates. I can't push these updates because the tip of the current branch is behind the remote. 

Does this mean I need to `git reset` the master branch too? Will that ruin/auto-close my API PR?

**Comment by cdeil** [comment]
@bmorris3 – Hmmm ... usually my suggestion would be to start with new branches and pull requests. But in case of #4 it would be good to keep that branch and pull request going. And as far as I know it's not possible to change the base or feature branch for Github pull requests (see e.g. [here](http://stackoverflow.com/a/10081072/498873)).

So my suggestion would be that you:
* make a tarball of your local repo directory before doing git reset or rebase commands, in case something goes wrong from which it's hard or impossible to recover.
* keep the master branch in your local repo and on your github fork for the API spec. If you have extra commits on there, you can `git reset --hard` to get rid of them.
* For the feature branch discussed here, make a new branch off of upstream master (i.e. not containing the commits from your local master branch) and then `git cherry-pick` the commits you already have to work on that (https://github.com/bmorris3/astroplan/commit/048d875cca9d19644ed11268e034d0f418dbc7b5) into that branch. Then make a pull request and check the commit and diff on Github.

Makes sense?

Of course this is only a suggestions, any other solution you come up with is OK.

Good luck!

**Comment by bmorris3** [comment]
Thanks for the git tutoring, I'm actually really thankful to be dealing with these git problems with the shelter of git-proficient mentors so I can learn. How do you make a new branch off of the upstream master without overwriting the current master?

**Comment by cdeil** [comment]
This is not the git workflow I use, but [this](http://stackoverflow.com/a/1783426/498873) should work
```
git fetch upstream
git checkout -b my-feature upstream/master
```
By upstream I mean the core Astropy repo on github, depending on how you cloned the repo and set up your remotes it might be called `origin`.  Use this to check:
```
git remote -v
```

**Comment by cdeil** [comment]
This has been implemented by @bmorris3  in #8.
We might need `skyfield` or `pyephem` on readthedocs in the future, but I guess we can add this when the need arises.
